connectors/tests/test_source.py

```diff
@@ -77,6 +95,56 @@ def test_default():
     assert c.get("dd", 1) == 1
 
 
+@pytest.mark.parametrize(
+    "type, required, default_value, value, expected_value",
+    [
+        # these include examples that would not normally validate
+        # that is because `.value` does care about validation, it only returns a value
+        # strings
+        ("str", True, "default", "input", "input"),
+        ("str", True, "default", None, None),
+        ("str", True, "default", "", ""),
+        ("str", False, "default", "input", "input"),
+        ("str", False, "default", None, "default"),
+        ("str", False, "default", "", "default"),
+        # integers
+        ("int", True, 3, 1, 1),
+        ("int", True, 3, None, None),
+        ("int", False, 3, 1, 1),
+        ("int", False, 3, None, 3),
+        # lists
+        ("list", True, ["1", "2"], ["3", "4"], ["3", "4"]),
+        ("list", True, ["1", "2"], [], []),
+        ("list", True, ["1", "2"], [""], [""]),
+        ("list", True, ["1", "2"], [None], [None]),
+        ("list", True, ["1", "2"], None, None),
+        ("list", False, ["1", "2"], ["3", "4"], ["3", "4"]),
+        ("list", False, ["1", "2"], [], ["1", "2"]),
+        ("list", False, ["1", "2"], [""], ["1", "2"]),
+        ("list", False, ["1", "2"], [None], ["1", "2"]),
```

Comment:
These surprise me. In most languages, while `[]` might be considered "empty" or "falsey", `[""]` and `[null]` would not, as the list is literally not empty (has a size of 1). I'm not sure that it matters, since IDK how you could possibly set a list like this from Kibana, but if someone was using an API or programmatic access with the protocol, this could be valid, and isn't the behavior I'd expect.

Reply:
I don't expect this to ever happen either, and Python doesn't consider these empty by default. I specifically wrote checks against these "empty-ish" list cases [here](https://github.com/elastic/connectors-python/blob/1267c8aebc59c2417728066e74dca80bda997cac/connectors/source.py#L109-L114) because someone could somehow add these by accident. Empty-ish lists would break things if they aren't validated properly.

That being said, this `.value` function doesn't care about validations, so it just returns whatever invalid value the user has added anyway.